mgmt/LocalManager.cc

```diff
@@ -924,19 +924,18 @@ LocalManager::startProxy(const char *onetime_options)
     int i = 0;
     char *options[32], *last, *tok;
     bool open_ports_p = false;
+    char real_proxy_options[16384]; // Arbitrary
 
-    Vec<char> real_proxy_options;
-
-    real_proxy_options.append(proxy_options, strlen(proxy_options));
+    ink_strlcpy(real_proxy_options, proxy_options, strlen(proxy_options));
     if (onetime_options && *onetime_options) {
-      real_proxy_options.append(" ", strlen(" "));
-      real_proxy_options.append(onetime_options, strlen(onetime_options));
+      ink_strlcat(real_proxy_options, " ", sizeof(real_proxy_options));
+      ink_strlcat(real_proxy_options, onetime_options, sizeof(real_proxy_options));
     }
 
     // Make sure we're starting the proxy in mgmt mode
-    if (strstr(proxy_options, MGMT_OPT) == 0 && strstr(onetime_options, MGMT_OPT) == 0) {
-      real_proxy_options.append(" ", strlen(" "));
-      real_proxy_options.append(MGMT_OPT, sizeof(MGMT_OPT) - 1);
+    if (strstr(real_proxy_options, MGMT_OPT) == 0) {
+      ink_strlcat(real_proxy_options, " ", sizeof(real_proxy_options));
```

Comment:
This might be better done by declaring earlier `static const char space = ' ';` and then using `ink_strlcat(real_proxy_options, &space, 1);`.